internal/controller/nificonnection_controller.go

```diff
@@ -685,6 +701,49 @@ func (r *NifiConnectionReconciler) DeleteConnection(ctx context.Context, clientC
 				}
 			}
 		}
+	} else if original.Spec.Source.Type == v1alpha1.ComponentProcessGroup {
+		// Retrieve NifiResource information
+		sourceInstance, err := k8sutil.LookupNifiResource(r.Client, original.Spec.Source.Name, original.Spec.Source.Namespace)
+		if err != nil {
+			return err
+		}
+
+		// Check Resource is Process Group
+		if sourceInstance.Spec.Type != v1.ResourceProcessGroup {
+			return errors.New("source resource is expected to be Process Group, but actual reasource is not")
```

Comment:
```suggestion
			return errors.New(fmt.Sprintf("Source resource %s in %s is expected to be Process Group",
				sourceInstance.Name, sourceInstance.Namespace))
```
To add more information, and make sure it's indicated that it's a `Source`and not a `Destination`.